src/browser/api_protocol/api_handlers/authorization.js

```diff
@@ -211,6 +211,25 @@ function AuthorizationApiHandler() {
 
     /*jshint unused:false */
     apiProtocolBase.registerActionMap(authenticationApiMap);
-}
+};
+
+const isConnectionAuthenticated = (msg, next) => {
+    const { data, nack, identity, strategyName } = msg;
+    const { uuid } = identity;
+    const action = data && data.action;
+    const isExternalConnection = strategyName === 'WebSocketStrategy';
+    const isAuthAction = authenticationApiMap.hasOwnProperty(action);
+    const isConnAuthed = ExternalApplication.getExternalConnectionByUuid(uuid);
+
+    // Prevent all API calls from unauthenticated external connections,
+    // except for authentication APIs
+    if (isExternalConnection && !isAuthAction && !isConnAuthed) {
+        return nack(new Error('This connection must be authenticated first'));
+    }
 
-module.exports.AuthorizationApiHandler = AuthorizationApiHandler;
+    next();
+};
+
+module.exports.registerMiddleware = function(requestHandler) {
+    requestHandler.addPreProcessor(isConnectionAuthenticated);
```

Comment:
this code executes for every request, can we short circuit it early, before we get the connectionByUuid.

Something like:

```
if (isExternalConnection) {
    next();
} else {
    //check if is authenticated and either next or nack.
}
```